stepconf/stepconf.go

```diff
@@ -86,7 +86,7 @@ func setField(field reflect.Value, value, constraint string) error {
 		field = field.Elem()
 	}
 
-	switch field.Kind() {
+	switch field.Kind() { //nolint:exhaustive
```

Comment:
Any reason we shouldn't be exhaustive here?

Reply:
This whole `stepconf` package is strongly tied to the step inputs declared in a `step.yml`. Step inputs at the moment can only be strings, bools, ints, floats and lists, so here in this code we don't expect any other type. Supporting other types would have to land in the CLI and the `step.yml` specification first.